### PR TITLE
Added the Inventory pickup item event

### PR DESCRIPTION
**Pull request by DeltaRays**
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 1]
```diff
@@ -561,11 +562,15 @@
...
-		.examples("on enchant:",
-			"\tif the clicked button is enchantment option 1:",
-			"\t\tset the applied enchantments to sharpness 10 and unbreaking 10")
-		.since("2.5");
+			.description("Called when a player successfully enchants an item.",
+				" To get the enchanted item, see the <a href='expressions.html#ExprEnchantEventsEnchantItem'>enchant item expression</a>")
+			.examples("on enchant:",
+				"\tif the clicked button is enchantment option 1:",
+				"\t\tset the applied enchantments to sharpness 10 and unbreaking 10")
+			.since("2.5");
```
Comment (ShaneBeee) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:570]: As stated previously in another PR, line changes should only be done for things relevant to the theme of the PR.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 2]
```diff
@@ -974,6 +975,28 @@ public Inventory get(final InventoryCloseEvent e) {
...
+			}
+		}, 0);
+		EventValues.registerEventValue(InventoryPickupItemEvent.class, Item.class, new Getter<Item, InventoryPickupItemEvent>() {
+			@Nullable
+			@Override
+			public Item get(InventoryPickupItemEvent event) {
+				return event.getItem();
+			}
+		}, 0);
+		EventValues.registerEventValue(InventoryPickupItemEvent.class, ItemStack.class, new Getter<ItemStack, InventoryPickupItemEvent>() {
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:993]: Use ItemType

**Review by FranKusmiruk — DISMISSED** [review]

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java (hunk 2 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:993]: Wouldn't there be issues with that?

Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:993]: You can do
```java
new ItemType(event.getItem().getItemStack());
```

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 3]
```diff
@@ -567,5 +568,9 @@
 			"\tif the clicked button is enchantment option 1:",
 			"\t\tset the applied enchantments to sharpness 10 and unbreaking 10")
 		.since("2.5");
+		Skript.registerEvent("Inventory Pickup", SimpleEvent.class, InventoryPickupItemEvent.class, "inventory pick[ ]up")
+			.description("Called when an inventory (a hopper, a hopper minecart, etc.) picks up an item")
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:572]: When making line breaks, Java standards require two indentations not a single indentation. You can look at the other events.

**Review by APickledWalrus — APPROVED** [review]
LGTM. Is it possible to modify the item though? Through Item#setItem for example (I think that's the method lol)

If it currently isn't possible, it might be useful to be able to do so.

**Review by ShaneBeee — APPROVED** [review]

**Review by ShaneBeee — APPROVED** [review]
looks good to me

**Review by ShaneBeee — APPROVED** [review]

**Review by APickledWalrus — APPROVED** [review]

**Comment by ShaneBeee** [comment]
Thank you for your contribution

**Pull request merged**